src/context.rs

```diff
@@ -34,10 +36,13 @@ impl<'a> Context<'a> {
             .map(|entry| entry.path())
             .collect::<Vec<PathBuf>>();
 
+        let repository: Option<Repository> = Repository::discover(&current_dir).ok();
```

Comment:
It looks like we are running `Context::new()` every time?
Perhaps this is premature but it could be a good idea to chuck the Context struct on the heap and return a reference instead? try to only it initialize once and then reuse across re-renders since i'm pretty sure its read-only anyway? 

Could make it easy to do the caching you were talking of but then we have to think about invalidation which is a whole thing. 

Anyway just a thought, something we could look into later.

Reply:
> It looks like we are running Context::new() every time?
> Perhaps this is premature but it could be a good idea to chuck the Context struct on the heap and return a reference instead?

I'm not sure I understand. `Context::new()` should only be running once in `print.rs` every time the whole prompt is being rendered. The only time context is used is by the reference passed here to generate the modules:
https://github.com/starship/starship/blob/master/src/print.rs#L30

I was hoping to eventually implement some form of caching that would persist between calls to `starship`, but yeah, that will end up being complicated.